library/src/main/kotlin/com/livefront/gsonkotlinadapter/KoltinReflectiveTypeAdapterFactory.kt

```diff
@@ -40,61 +40,56 @@ class KotlinReflectiveTypeAdapterFactory private constructor(
-
-        private val declaringClass: Class<T> = primaryConstructor.javaConstructor?.declaringClass!!
-
-        /**
-         * Provides a mapping between parameter and automatic type-specific defaults (i.e. null for
-         * objects, 0 for numeric primitives, '\u0000' for chars, false for booleans). If the
-         * parameter [KParameter.isOptional] then it will not exist in this map and the
-         * manually-supplied default value will be used when the constructor is called. The
-         * contents of this map may vary based on the [enableDefaultPrimitiveValues] property.
-         */
-        private val constructorParameterDefaultsMap: Map<KParameter, Any?> = primaryConstructor
-            .parameters
-            .filterNot(KParameter::isOptional)
-            .associateWith { if (enableDefaultPrimitiveValues) it.defaultValue else null }
 
-        private val constructorParameterNameMap: Map<KParameter, List<String>> = primaryConstructor
-            .parameters
-            .map { it to it.getSerializedNames(declaringClass) }
-            .toMap()
+        val primaryConstructor: KFunction<T> = kotlinRawType
+            .primaryConstructor
+            ?.apply { isAccessible = true }
+            ?: return null
+        val declaringClass: Class<T> = primaryConstructor.javaConstructor!!.declaringClass!!
+        val innerAdapters: MutableMap<KParameter, TypeAdapter<*>> = mutableMapOf()
+        val constructorParameterDefaultsMap: MutableMap<KParameter, Any?> = mutableMapOf()
+        val invalidReadParameters: MutableSet<KParameter> = mutableSetOf()
+        val constructorMap: MutableMap<String, KParameter> = mutableMapOf()
 
-        private val invalidReadParameters: Set<KParameter> = constructorParameterNameMap
-            .filter { (parameter, names) ->
-                if (enableDefaultPrimitiveValues && parameter.isPrimitive) {
-                    false
+        primaryConstructor.parameters.forEach {
+            val names: List<String> = it.getSerializedNames(declaringClass)
+            if (names.isNotEmpty()) {
+                innerAdapters[it] = gson.getAdapter(type.resolveParameterType(it))
+            }
+            if (!it.isOptional) {
+                constructorParameterDefaultsMap[it] = if (enableDefaultPrimitiveValues) {
+                    it.defaultValue
                 } else {
-                    names.isEmpty() && !parameter.isOptional
+                    null
                 }
-            }
-            .keys
 
-        private val constructorMap: Map<String, KParameter> = constructorParameterNameMap
-            .entries
-            .flatMap { (parameter, names) -> names.map { it to parameter } }
-            .toMap()
+                if (names.isEmpty() && (!it.isPrimitive || !enableDefaultPrimitiveValues)) {
+                    invalidReadParameters += it
+                }
+            }
+            names.forEach { name -> constructorMap[name] = it }
```

Comment:
This whole block of code is large enough that you might consider explicitly naming the `it` to `parameter`.

Reply:
Sure